tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/experimental/plugins/KotlinNativeBasePlugin.kt

```diff
@@ -0,0 +1,75 @@
+package org.jetbrains.kotlin.gradle.plugin.experimental.plugins
+
+import org.gradle.api.Plugin
+import org.gradle.api.component.SoftwareComponentContainer
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.internal.FeaturePreviews
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.internal.tasks.TaskContainerInternal
```

Comment:
❌ You are using internal Apis here.
Is that expected?

Reply:
Unfortunately yes. I try to keep the plugin close to the new Gradle model for native projects (e.g. see [this blog post](https://blog.gradle.org/introducing-the-new-cpp-plugins)) and maximally reuse a code provided by it. This native support in Gradle is at a preview stage so there is no public wrappers for many of these APIs.